### PR TITLE
Add support for installing to Scenarios

### LINKED ISSUE #1358: Allow installs to saves/scenarios
Scenarios are different from tutorials, and include things like the asteroid redirect mission.

This would require a new install target for `install_to` and associated code, and would be pretty much identical to the tutorials target except for the path.

### PR DESCRIPTION
**Pull request by pjf**
Closes #1358.

This feels like it was really easy to add, but it's a fairly straightforward change.

### CODE REVIEW

**Comment by pjf** [comment]
Using:

```
{
    "spec_version": "v1.14",
    "identifier": "RaceKerbol",
    "name": "Race Kerbol",
    "abstract": "Race Kerbol (the sun) in a series of scenarios from start to finish while trying to keep as close to the ground as possible to get a higher score! Your craft is solar powered, so make sure to keep your panels exposed to the sun as much as possible.",
    "license": "CC-BY-NC-SA",
    "comment": "Contains both GPL and CC-BY-NC-SA sections",
    "ksp_version": "1.0.4",
    "depends": [
        {
            "name": "ModuleManager"
        },
        {
            "name": "KerbalFoundries"
        }
    ],
    "install": [
        {
            "file": "saves/scenarios",
            "install_to": "Scenarios"
        },
        {
            "file": "GameData/RaceKerbol",
            "install_to": "GameData"
        }
    ],
    "version": "v1.0",
    "author": "Lyneira",
    "download": "https://github.com/Lyneira/RaceKerbol/releases/download/v1.0/RaceKerbol-1.0.zip",
    "resources": {
        "repository": "https://github.com/Lyneira/RaceKerbol"
    },
    "download_size": 50974,
    "x_generated_by": "netkan"
}
```

results in:

```
$ ckan.exe show RaceKerbol
Race Kerbol: Race Kerbol (the sun) in a series of scenarios from start to finish while trying to keep as close to the ground as possible to get a higher score! Your craft is solar powered, so make sure to keep your panels exposed to the sun as much as possible.

Module info:
- version:	v1.0
- authors:	Lyneira
- status:	stable
- license:	CC-BY-NC-SA

Depends:
- ModuleManager
- KerbalFoundries

Provides:
- RaceKerbol

Resources:
- repository: https://github.com/Lyneira/RaceKerbol

Filename: 79712415-RaceKerbol-v1.0.zip

Showing 13 installed files:
- saves/scenarios/scenarios
- saves/scenarios/scenarios/Race Kerbol 01 - Westbound.sfs
- GameData/RaceKerbol
- GameData/RaceKerbol/LandDistance.dll
- GameData/RaceKerbol/LD.png
- GameData/RaceKerbol/Parts
- GameData/RaceKerbol/Parts/atmoPlasmaEngine.cfg
- GameData/RaceKerbol/Parts/radialFlatSolarPanelx16.cfg
- GameData/RaceKerbol/Parts/radialFlatSolarPanelx4.cfg
- GameData/RaceKerbol/Parts/Repulsor.cfg
- GameData/RaceKerbol/PluginData
- GameData/RaceKerbol/PluginData/LandDistance
- GameData/RaceKerbol/PluginData/LandDistance/config.xml
```

Which looks pretty good to me!

**Comment by plague006** [comment]
:+1:

**Comment by pjf** [comment]
Oh, travis doesn't know we're going to bump the spec version. Let me fix that. :)

File: Core/ModuleInstaller.cs [hunk 1]
```diff
@@ -452,10 +452,17 @@ internal static List<InstallableFile> FindInstallableFiles(ModuleInstallDescript
                     installDir = ksp == null ? null : ksp.Tutorial();
                     makeDirs = true;
                     break;
+
+                case "Scenarios":
```
Comment (Postremus) [inline Core/ModuleInstaller.cs]: Does three cases look the same. maybe merge them together into one method?

**Comment by Lyneira** [comment]
Those results do look like what should happen when installing Race Kerbol. :)

File: Core/ModuleInstaller.cs (hunk 1 above)
Comment (pjf) [inline Core/ModuleInstaller.cs]: @Postremus : I'm not sure that would reduce our complexity at all (one has makeDirs false, two have makrDirs true, all install to different locations). But I'd welcome a commit that simplifies things if you have one! (I made sure to do this in the KSP-CKAN repo for easy adjustments)

**Comment by pjf** [comment]
And merge conflicts all fixed. :)

**Comment by pjf** [comment]
Now looking for my fabulous contributors to review and merge this if it looks good! (Direct puppy-dog eyes at @Postremus, @dbent, and @mgsdk...)

File: Core/ModuleInstaller.cs (hunk 1 above)
Comment (Postremus) [inline Core/ModuleInstaller.cs]: You are right.

**Comment by Postremus** [comment]
Nothing against a merge, but PR this needs to be rebased.

**Comment by pjf** [comment]
@Postremus : Done! :)

**Comment by pjf** [comment]
And updated with a fix that should stop [this failure](http://puu.sh/jxcg3/995f343877.png) happening under Windows. (This was caused by a buggy test.)

**Comment by Postremus** [comment]
Tested, works. merged with thanks.

@Lyneira 
You may wan't to change your .netkan slightly. It currently installs into
```
saves/scenarios/scenarios/Race Kerbol 01 - Westbound.sfs
```
This may cause issues, don't know.

**Pull request merged**